lightning/src/ln/channelmanager.rs

```diff
@@ -10357,9 +10357,21 @@ where
 	#[cfg(c_bindings)]
 	create_refund_builder!(self, RefundMaybeWithDerivedMetadataBuilder);
 
+	/// Retrieve our cached [`Offer`]s for receiving async payments as an often-offline recipient.
+	/// Will only be set if [`UserConfig::paths_to_static_invoice_server`] is set and we succeeded in
+	/// interactively building a [`StaticInvoice`] with the static invoice server.
+	#[cfg(async_payments)]
+	pub fn get_cached_async_receive_offers(&self) -> Vec<Offer> {
```

Comment:
These pub fns added, do they need a high level test?

Also curious to see how this is going to be used in ldk node.

Reply:
This is tested in #3628. As discussed offline, I looked into migrating some of the testing to this PR by using mocks for the server. It doesn't look particularly trivial honestly, we'd still need a lot of the code that's in the follow-up for unwrapping + sending messages and creating blinded paths on behalf of the server. 

I'm still going to try to get you something to look at on the ldk-node front though 🫡 hoping to have some updates soon.

Reply:
Would the order have mattered here? If the server pr would have been the first, would it have been easier to mock the client? Not asking for this, to be clear. Just curious.